### PR TITLE
Add a doc for git best practices

### PR DESCRIPTION
**Pull request by scorinaldi**
This PR adds a doc laying out git best practices. I'd love to get feedback on which of these folks like and want to adopt , and which I've left out.

### CODE REVIEW

File: .github/workflows/test_pr.yml [hunk 1]
```diff
@@ -45,7 +45,6 @@ jobs:
           -u ${{ env.PROJECT_TOKEN }} \
```
[the remark this answers is not in the document]
Comment (itsjoeoui) [reply .github/workflows/test_pr.yml:45]: Let's just leave it here! Looks good.

**Review by itsjoeoui — APPROVED** [review]

File: .github/workflows/test_pr.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (scorinaldi) [reply .github/workflows/test_pr.yml:45]: SG! ty

File: git_best_practices.md
[the remark this answers is not in the document]
Comment (scorinaldi) [reply git_best_practices.md:1]: Sounds good! Up next I will add it to the pr templates so folks can see it as they make their PRs.

File: git_best_practices.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Git Best Practices
+
+---
+
+Here are a few easy ways to make sure your PR gets reviewed quickly, and your code gets merged as efficiently as possible.
+
+## Scope your branches!
+
+- To avoid name collsions please prefix your branch names with your name (e.g. use `sharif-my-cool-feature` not `my-cool-feature`)
```
Comment (dev-nested) [inline git_best_practices.md:9]: for naming we could use `/` to separate the dev and branch name eg `sharif/<branch_name>` too, then these branches will show up as collapsible by dev name

**Pull request merged**